app/models/upload.rb

```diff
@@ -72,6 +72,18 @@ def self.last_uploads_rows
     uploads.sort_by { |upload| upload.csv_type.downcase }
   end
 
+  def self.latest_uploads
+    preview = Version.current_preview
+    production = Version.current_production
+    return if production.blank?
+
+    created_at = preview.present? && preview.number == production.number ? production.created_at : preview.created_at
```

Comment:
What is supposed to happen if `preview.present?` returns `false`?

Reply:
Fixed logic